dandi/cli/cmd_service_scripts.py

```diff
@@ -0,0 +1,103 @@
+from difflib import unified_diff
+import json
+from pathlib import PurePosixPath
+
+import click
+from dandischema.consts import DANDI_SCHEMA_VERSION
+from packaging.version import Version
+
+from .base import map_to_click_exceptions
+from .. import lgr
+from ..dandiapi import RemoteBlobAsset
+from ..dandiarchive import parse_dandi_url
+from ..exceptions import NotFoundError
+
+
+@click.group()
+def service_scripts() -> None:
+    """Various utility operations"""
+    pass
+
+
+@service_scripts.command()
+@click.option("--diff", is_flag=True, help="Show diffs of old & new metadata")
+@click.option(
+    "--when",
+    type=click.Choice(["newer-schema-version", "always"]),
+    default="newer-schema-version",
+    help="When to re-extract an asset's metadata",
+    show_default=True,
+)
+@click.argument("url")
+@map_to_click_exceptions
+def reextract_metadata(url: str, diff: bool, when: str) -> None:
+    """
+    Recompute & update the metadata for NWB assets on a remote server.
+
+    Running this command requires the fsspec library to be installed with the
+    `http` extra (e.g., `pip install "fsspec[http]"`).
+    """
+    from ..metadata import nwb2asset  # Avoid heavy import at top level
+
+    parsed_url = parse_dandi_url(url)
+    if parsed_url.version_id not in (None, "draft"):
+        raise click.UsageError("URL cannot point to a published Dandiset")
+    parsed_url.version_id = "draft"
+    current_schema_version = Version(DANDI_SCHEMA_VERSION)
+    with parsed_url.navigate(authenticate=True, strict=True) as (_, _, assets):
+        for asset in assets:
+            if PurePosixPath(asset.path).suffix.lower() != ".nwb":
+                lgr.info(
+                    "Asset %s (%s) is not NWB; skipping", asset.identifier, asset.path
+                )
+            assert isinstance(asset, RemoteBlobAsset)
+            lgr.info("Processing asset %s (%s)", asset.identifier, asset.path)
+            if when == "always":
+                do_reextract = True
+            else:
+                try:
+                    sv = asset.get_raw_metadata()["schemaVersion"]
+                except KeyError:
+                    do_reextract = True
+                else:
+                    schemaVersion = Version(sv)
+                    if schemaVersion < current_schema_version:
+                        do_reextract = True
+                    elif schemaVersion == current_schema_version:
+                        lgr.info(
+                            "Asset's schemaVersion is up to date; not reextracting"
+                        )
```

Comment:
```suggestion
                            "Asset's schemaVersion %s is up to date; not reextracting",
                            schemaVersion
                        )
```
since ATM I am puzzled as how I see this whenever `dandi ls` gives me 0.4.4 for the asset:

```
❯ dandi ls --metadata assets dandi://dandi/000027/sub-RAT123/sub-RAT123.nwb | grep schemaVersion
    schemaVersion: 0.4.4
2023-03-02 17:11:02,367 [    INFO] Logs saved in /home/yoh/.cache/dandi-cli/log/20230302221101Z-3397824.log
❯ less /home/yoh/.cache/dandi-cli/log/20230302221035Z-3397727.log
❯ dandi service-scripts reextract-metadata --diff --when newer-schema-version dandi://dandi/000027/sub-RAT123/sub-RAT123.nwb
2023-03-02 17:11:35,291 [    INFO] Processing asset 838bab7b-9ab4-4d66-97b3-898a367c9c7e (sub-RAT123/sub-RAT123.nwb)
2023-03-02 17:11:35,345 [    INFO] Asset's schemaVersion is up to date; not reextracting
2023-03-02 17:11:35,348 [    INFO] Logs saved in /home/yoh/.cache/dandi-cli/log/20230302221132Z-3397999.log
```
and even though the actual reason is that we `ls` the released version
```
❯ dandi ls --metadata assets dandi://dandi/000027/sub-RAT123/sub-RAT123.nwb | grep schemaVersion
    schemaVersion: 0.4.4
```
```
❯ dandi ls --metadata assets dandi://dandi/000027@draft/sub-RAT123/sub-RAT123.nwb | grep schemaVersion
    schemaVersion: 0.6.3
```